src/org/usfirst/frc/team930/robot/controller/Controller.java

```diff
@@ -0,0 +1,72 @@
+package org.usfirst.frc.team930.robot.controller;
+
+import java.util.TimerTask;
+/**
+ * Abstract class to control inputs, or in its basic form
+ * an enable-able looped calculator. 
+ * 
+ * @author Nick Janovetz
+ * @version 2016.02.03_1 Formatting changes
+ *
+ */
+abstract class Controller {
+
+	private static final double DEFAULT_PERIOD = .05;
+	private boolean enabled;
+	private java.util.Timer loop;
```

Comment:
So the Timer class runs on the JVM layer which, for our purposes, is more inaccurate than we want. This was one of our issues with the PID code last year.

2016, they've added a timer that runs on the FPGA that we can utilize. It's pretty similar to what you've got so it won't require too much refactoring. Called the Notifier class - http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/Notifier.html